modules/programs/zsh.nix

```diff
@@ -168,9 +189,21 @@ let
         type = types.str;
         example = "$HOME/my_customizations";
         description = ''
-          Path to a custom oh-my-zsh package to override config of
+          Path to a custom oh-my-zsh directory to override config of
           oh-my-zsh. See <https://github.com/robbyrussell/oh-my-zsh/wiki/Customization>
           for more information.
+          Cannot be used with `customPkgs`.
```

Comment:
```nix
    assertions = let
      customPkgs = cfg.oh-my-zsh.customPkgs;
      custom = cfg.oh-my-zsh.custom;
    in [{
      assertion = custom != "" && length customPkgs;
      message = "Both `programs.zsh.oh-my-zsh.custom` ('" + custom + "') and `programs.zsh.oh-my-zsh.customPkgs` ([" + toString customPkgs + "]) set."
    }]
```
I'm not quite sure where to put this, because the config attribute is structure a little differently in `zsh.nix`.

Reply:
I think like this since it should be `config.assertions`:
```nix
config = mkIf cfg.enable (mkMerge [
  {
   assertions = /* ... */;
  }
  # ...
])
```

Reply:
I pushed a commit on my fork to the branch `bad-omz` that should have the assertion done. However, I'm getting a ton of warnings when running `nix develop --ignore-environment .#all`, complaining that there's `No newline at end of file` ... but I didn't touch the end of the file, and it doesn't look like there were these errors before? Help would be appreciated.

Reply:
I ran it locally and the output is:
```
--- actual
+++ expected
@@ -1 +0,0 @@
-Both `programs.zsh.oh-my-zsh.custom` ('') and `programs.zsh.oh-my-zsh.customPkgs` ([]) set.
\ No newline at end of file
For further reference please introspect /nix/store/10mp1xvkn6m7fjb508x53fsxpirfixhj-nmt-report-zsh-history-path-old-default
zsh-history-substring-search: FAILED
Expected home-files/asserts/assertions.actual to be same as /nix/store/hkmk0rdn9bwsnmnpdsdbw7gm4har8jwz-assertions.expected but were different:
```

The important part is `-Both [...]`. Basically your warning is being triggered by all the tests.  
I pushed a fixed assert with a couple other tweaks and a test for it in my fork as `bad-omz`. You can cherry-pick that.

Reply:
Well, that was an embarrassing mistake; thank you for catching that. Fix should be pushed onto this branch!

Reply:
TBH I almost missed it too cause I was looking for the newline thing :smile: 

Resolving this thread since the assert is in and tested.